src/mlagility/api/model_api.py

```diff
@@ -116,10 +123,71 @@ def benchmark_model(
                 )
                 perf = cpu_model.benchmark(backend=backend)
 
+        elif device == "x86" and runtime in ["torch", "torch_compiled"]:
+
+            # Create cache folder with stats file
+            # Although building with an empty sequence is possible, we don't want
+            # the model to be rebuilt when other devices are used, causing the
+            # results inside of the stats file to be lost.
+            build_model(
+                model=model,
+                inputs=inputs,
+                build_name=build_name,
+                cache_dir=cache_dir,
+                rebuild=rebuild,
+                sequence=sequence,
+            )
```

Comment:
Trying to understand what's happening here... are you building the ONNX files even though this code path doesn't use them?

Reply:
Yeah, if we create a build without creating onnx files for `torch` and then try to run `ort`, for example, we will not have a cache hit due to the different sequences. I will investigate on monday if we can get a better solution for this.

Reply:
Solved! Exporting to ONNX is not needed when runtimes `torch-eager` or `torch-compiled` are used.